.gitignore

```diff
@@ -1,13 +1,25 @@
+# Android Studio generated
 *.iml
 .gradle
-local.properties
-.idea/
+/local.properties
+/.idea/caches
+/.idea/libraries
+/.idea/modules.xml
+/.idea/workspace.xml
+/.idea/navEditor.xml
+/.idea/assetWizardSettings.xml
```

Comment:
All of these are covered below by `local.properties` and `/.idea`, and the latter seem better.

```suggestion
```

Reply:
I figured there was value in making explicit what the upstream Android Studio gitignore is. I take it you disagree?

Reply:
That does make sense. But I think it's confusing like this. If I see a specific rule matching `/local.properties` I will be very surprised to find a more general rule later in the file that excludes that file in every directory, not just the root.

If the intent is to document the upstream gitignore, maybe comment those out and add a comment that they are commented out because of later rules but there for reference?

Reply:
The `/local.properties` and `local.properties` both being in that list is upstream. I did not edit any of that.

```
*.iml
.gradle
/local.properties
/.idea/caches
/.idea/libraries
/.idea/modules.xml
/.idea/workspace.xml
/.idea/navEditor.xml
/.idea/assetWizardSettings.xml
.DS_Store
/build
/captures
.externalNativeBuild
.cxx
local.properties
```

So that doesn't really make sense for the part of your argument of "it's confusing to list both upstream and your rules like that".

Reply:
Eh... that just makes upsteam the main cause of the confusion. It's still confusing IMO.

Reply:
I don't think "upstream made it confusing" is an argument for not removing said confusion (and making sure not to add to it) :)